swarm/storage/resource.go

```diff
@@ -465,78 +447,86 @@ func parseUpdate(blob []byte) (period uint32, version uint32, ensname []byte, da
 // A resource update cannot span chunks, and thus has max length 4096
 func (self *ResourceHandler) Update(name string, data []byte) (Key, error) {
 
-	ok, err := self.validator.isOwner(name)
-	if err != nil {
-		return nil, err
-	} else if !ok {
-		return nil, fmt.Errorf("Not owner of '%s'", name)
+	var sigoffset int
+	if self.validator != nil {
+		sigoffset = signatureLength
 	}
 
-	// can be only one chunk long minus 65 byte signature
-	if int64(len(data)) > self.maxChunkData {
-		return nil, fmt.Errorf("Data overflow: %d / %d bytes", len(data), 4096-signatureLength)
+	// get the cached information
+	rsrc := self.getResource(name)
+	if rsrc == nil {
+		return nil, fmt.Errorf("Resource object not in index")
+	}
+	if !rsrc.isSynced() {
+		return nil, fmt.Errorf("Resource object not in sync")
 	}
 
-	// get the cached information
-	self.resourceLock.Lock()
-	defer self.resourceLock.Unlock()
-	resource, ok := self.resources[name]
-	if !ok {
-		return nil, fmt.Errorf("No such resource")
-	} else if resource.updated.IsZero() {
-		return nil, fmt.Errorf("Invalid resource")
+	// an update can be only one chunk long
+	datalimit := self.chunkSize() - int64(sigoffset-len(name)-8)
```

Comment:
Since you only use `sigoffset` here to calculate the data limit, I would move that here and extract it into a function and also comment why is it calculated like this.
Something like
```
func (self *ResourceHandler) getDataLimit() int64 {
	var sigoffset int
	if self.validator != nil {
		sigoffset = signatureLength
	}
        return self.chunkSize() - int64(sigoffset-len(name)-8)
}
```


self.chunkSize() - int64(sigoffset-len(name)-8)